.evergreen/find-python3.sh

```diff
@@ -291,3 +291,32 @@ find_python3() (
 
   return 0
 )
+
+#
+# Usage:
+#   ensure_python3
+#   PYTHON_BINARY=$(ensure_python3)
+#   PYTHON_BINARY=$(ensure_python3 2>/dev/null)
+#
+# If successful, print the name of the binary stdout (pipe 1).
+# Otherwise, no output is printed to stdout (pipe 1).
+#
+# Diagnostic messages may be printed to stderr (pipe 2). Redirect to /dev/null
+# with `2>/dev/null` to silence these messages.
+#
+# If PYTHON or PYTHON_BINARY are set, it will return that value.  Otherwise
+# it will use find_python3 to return a suitable value.
+#
+ensure_python3() {
+  declare python_binary
+  python_binary="${PYTHON:-"${PYTHON_BINARY:-""}"}"
+  if [ -z "${python_binary}" ]; then
+    echo "Finding Python3 binary..." 1>&2
+    python_binary="$(find_python3 2>/dev/null)" || return
+    echo "$python_binary"
+    echo "Finding Python3 binary... done." 1>&2
+  else
+    echo "Using Python binary $PYTHON" 1>&2
+    echo "$python_binary"
+  fi
```

Comment:
```suggestion
  {
    if [ -z "${python_binary}" ]; then
      echo "Finding Python3 binary..."
      python_binary="$(find_python3 2>/dev/null)" || return
      echo "Finding Python3 binary... done."
    else
      echo "Using Python binary $PYTHON"
    fi
  } 1>&2
  echo "${python_binary:?}"
```
This may be a bit more convenient form to separate stdout vs. stderr redirection. (The `1>&2` can be applied to the `if` itself, but using `{}` is probably easier to understand.)